### PR TITLE
Use Versions.props to eliminate source-build prebuilts

### PR DESCRIPTION
**Pull request by MichaelSimons**
Related to https://github.com/dotnet/source-build/issues/2835

### Context

Update Packages.props to utilizes properties defined in versions.props which get overridden by the built versions in source-build.  This eliminates prebuilts.  Before these changes, references to 6.0.0 were being added instead of the current versions e.g. System.Text.Json,6.0.4.  

I applied the same change to all 6.0.0 references.  This is only a problem for System.Text.Json but will be once the other packages are serviced for 6.0.

### Testing

Ran a complete source-build to validate the prebuilts were removed.

### CODE REVIEW

**Review by rainersigwald — APPROVED** [review]

**Comment by MichaelSimons** [comment]
@rainersigwald - Can you please merge unless you expect others to review.  TIA

**Review by Forgind — APPROVED** [review]

**Comment by rainersigwald** [comment]
It should already be in 4xx; for 3xx target `vs17.2` (and we'll need tactics approval + a special exception from the VS QB to not ship the resulting MSBuild for that).

**Comment by MichaelSimons** [comment]
> a special exception from the VS QB to not ship the resulting MSBuild for that

We can carry a source-build patch in 3xx since this is only about removing prebuilts.

**Comment by rainersigwald** [comment]
That would be a lot easier for us 😇

**Pull request merged**
